Java/core/src/main/java/com/amazon/randomcutforest/store/SmallIndexManager.java

```diff
@@ -19,36 +19,54 @@
 import static com.amazon.randomcutforest.CommonUtils.checkNotNull;
 import static com.amazon.randomcutforest.CommonUtils.checkState;
 
+import java.util.Arrays;
 import java.util.BitSet;
 import java.util.HashSet;
 import java.util.Set;
 
 /**
  * This class defines common functionality for Store classes, including
- * maintaining the stack of free pointers.
+ * maintaining the stack of free pointers. The stack may be implicitly defined.
+ * if freeIndexPointer exceeds freeIndexes.length, then any intermediate
+ * location say i, must contain entry capacity - i - 1
```

Comment:
These are useful implementation details, but I don't think that they will help people understand how to use the class. Maybe we could move this to an internal comment somewhere.

Reply:
Done.